code/game/objects/items/devices/taperecorder.dm

```diff
@@ -224,22 +226,9 @@
 			playsleepseconds = 1
 		i++
 
-	stop()
-
-
-/obj/item/taperecorder/attack_self(mob/user)
-	if(!mytape || mytape.ruined)
-		return
-	if(recording)
-		stop()
-	else
-		record()
-
-
-/obj/item/taperecorder/verb/print_transcript()
-	set name = "Print Transcript"
-	set category = "Object"
+	stop(TRUE)
 
+/obj/item/taperecorder/proc/print_transcript()
 	if(usr.incapacitated())
```

Comment:
I notice that every single one of these action procs are checking `usr.incapacitated` when this check could probably be moved out of these procs to the place where they're being called from.

Reply:
I believe these have to be here just in case they somehow fall asleep in the middle of choosing

Reply:
I believe these have to be here just in case they somehow fall asleep in the middle of choosing

Reply:
that's not what I'm pointing out. You call `usr.incapacitated()` in 3-4 procs when you can really just put one before the switch statement reading the results from the radial menu which will do exactly what you're trying to do with this check and keep the check scoped to the right proc.

Reply:
Ah, I see what you mean now